### ISSUE TITLE
oj_slash_string should either alloc each time or be frozen

### ISSUE THREAD

**Issue by enebo** [ohler55/oj#492]
Howdy,  I have been working on JRuby port of oj and I thought I would start reporting things I notice along the way.  oj_slash_string is returned in various places but it is a mutable string.  So if I modify it then all future returns will be the modified contents:

```ruby
requite 'oj'

doc = Oj::Doc.open("[1,2]")
b = doc.where?
b << "heh"
p doc.where?'
```

Semantically returning a new instance (rb_str_dup) would probably have least consequence but making it frozen would be another possible solution (although semantically all paths probably should then be frozen as well -- I could see this having a bigger possible effect on backwards compat).

**Comment by ohler55** [comment]
Good catch. It will be fixed in the next release which is going out today if there are no issues with the release process.

**Comment by ohler55** [comment]
Released with the fix. Slash is now frozen.

**Comment by ohler55** [comment]
Can this be closed?

**Comment by enebo** [comment]
@ohler. Sorry I figured you would just close it.  Seems reasonable to me. If people blindly gsub! Or something on return path maybe it causes issue but then again I was first person to report it and I was not doing that :)

**Comment by ohler55** [comment]
You did see that I put in a fix, right? Strings that are reused are now frozen.

**Comment by enebo** [comment]
Yeah. Seems reasonable
